src/App.tsx

```diff
@@ -1,24 +1,250 @@
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { TodoNotification } from './components/TodoNotification';
+import {
+  getTodos,
+  createTodo,
+  updateTodo,
+  deleteTodo,
+} from './api/todos';
+import { TodoStatus } from './types/TodoStatus';
+import { Todo } from './types/Todo';
+import { TodoListContext } from './context/TodoListContext';
 
-const USER_ID = 0;
+const USER_ID = 10509;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(TodoStatus.All);
+  const [hasError, setHasError] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [completedTodoIdList, setCompletedTodoIdList] = useState<number[]>([]);
+
+  const showErrorMessage = (message: string) => {
+    setHasError(true);
+    setErrorMessage(message);
+  };
+
+  const handleCloseButton = useCallback(() => {
+    setHasError(!hasError);
+  }, [hasError]);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      setHasError(false);
+    }, 3000);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [showErrorMessage]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todoList => setTodos(todoList))
+      .catch(() => showErrorMessage('Unable to load todos'));
+  }, [USER_ID]);
+
+  const visibleTodos = useMemo(() => {
+    let visibleTodoArray = todos;
+
+    switch (filterBy) {
+      case TodoStatus.Active:
+        visibleTodoArray = todos.filter(todo => !todo.completed);
+        break;
+
+      case TodoStatus.Completed:
+        visibleTodoArray = todos.filter(todo => todo.completed);
+        break;
+
+      case TodoStatus.All:
+      default:
+        break;
+    }
+
+    return visibleTodoArray;
+  }, [todos, filterBy]);
+
+  const activeTodosCount = useMemo(() => (
+    todos.filter(todo => !todo.completed).length
+  ), [visibleTodos]);
+
+  const areAllTodosCompleted = useMemo(() => (
+    todos.every(todo => todo.completed)
+  ), [todos]);
+
+  const toggleAllTodos = useCallback(async () => {
+    const todosToToggle = areAllTodosCompleted
+      ? todos.filter(todo => todo.completed)
+      : todos.filter(todo => !todo.completed);
+
+    setCompletedTodoIdList(todosToToggle.map(todo => todo.id));
+
+    try {
+      const updatedTodos = await Promise.all(todosToToggle
+        .map(todo => updateTodo(todo.id, { completed: !todo.completed })));
+
+      setTodos(currTodos => currTodos.map(todo => {
+        const updatedTodo = updatedTodos.find(task => task.id === todo.id);
+
+        return updatedTodo || todo;
+      }));
+    } catch {
+      showErrorMessage('Unable to update the todos');
+    } finally {
+      setCompletedTodoIdList([]);
+    }
+  }, [todos]);
+
+  const handleEnterKeyPress = useCallback((title: string) => {
+    setTempTodo({
```

Comment:
It's better to use a unique identifier for the todo id instead of using Math.random(). As a variant, consider installing and using a library like 'uuid' to generate unique ids